src/network/gossip.rs

```diff
@@ -297,11 +310,9 @@ impl SnapchainGossip {
     }
 
     pub async fn check_and_reconnect_to_bootstrap_peers(&mut self) {
-        let connected_peers_count = self.swarm.connected_peers().count();
-        // If we have no connected peers, try to reconnect to bootstrap peers
-        if connected_peers_count == 0 && !self.bootstrap_addrs.is_empty() {
```

Comment:
I think we should keep this behavior for read nodes? And only use the new one for validators. It's not required for read nodes to be connected to all validators